src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -265,9 +261,12 @@ private void launchNode(SchedulerDriver driver, Offer offer,
       TaskID taskId = TaskID.newBuilder()
           .setValue(String.format("task.%s.%s", taskName, taskIdName))
           .build();
+      //for name nodes, we need to get its own unique task name
```

Comment:
The chooseNodeName method should be moved to LiveState. Instead of having additional data structures we should instead use the list of running tasks in LiveState. If you could resolve the TODO storing the NN, JN and DN's separately, then we simply need to loop through the JN and NN list and assign the smallest number that isn't taken. Number of NN's and JN's will be small so this should be quick and we can avoid using the new data structures storing the nameNode names.